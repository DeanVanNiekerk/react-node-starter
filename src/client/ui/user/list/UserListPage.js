import React, { Component } from 'react';
import { connect } from 'react-redux'
import { Table } from 'reactstrap';

import Loader from '@/ui/common/Loader'
import Error from '@/ui/common/Error'
import { fetchUsers } from '@/state/users/list/actions'

class UserListPage extends Component {

  componentDidMount() {

    this.props.dispatch(fetchUsers());
  }

  render() {

    if (this.props.error)
      return <Error />
    if (this.props.fetching)
      return <Loader text="loading users..." />

    return (
      <div>
        <h3>Users</h3>
        <Table>
          <thead>
            <tr>
              <th>First Name</th>
              <th>Last Name</th>
            </tr>
          </thead>
          <tbody>
                {this.props.users.map(user =>
                    <tr key={user.id}>
                        <td>{user.firstName}</td>
                        <td>{user.lastName}</td>
                    </tr>
                )}
            </tbody>
        </Table>
      </div>
    )

  }
}

const mapStateToProps = state => ({
  users: state.users.list.items,
  fetching: state.users.list.fetching,
  error: state.users.list.error
})

export default connect(mapStateToProps)(UserListPage)