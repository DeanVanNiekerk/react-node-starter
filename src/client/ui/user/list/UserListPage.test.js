import React from 'react';
import UserListPage from './UserListPage';
import configureMockStore from 'redux-mock-store'
import renderer from 'react-test-renderer';

const mockStore = configureMockStore()

describe('UserListPage', () => {

    it('renders correctly', () => {

        const store = mockStore({ 
            users: {
                list: {
                    items: [
                        { id: 1, firstName: 'Jack', lastName: 'Johnson' }
                    ]
                }
            }
        })

        store.dispatch = jest.fn();

        const tree = renderer
            .create(<UserListPage store={store} />)
            .toJSON();

        expect(store.dispatch.mock.calls.length).toBe(1);

        expect(tree).toMatchSnapshot();
    });

});