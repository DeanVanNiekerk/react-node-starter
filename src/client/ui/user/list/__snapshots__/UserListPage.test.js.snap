// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserListPage renders correctly 1`] = `
<div>
  <h3>
    Users
  </h3>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          First Name
        </th>
        <th>
          Last Name
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Jack
        </td>
        <td>
          Johnson
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
